src/utils.ts

```diff
@@ -21,3 +25,84 @@ export const remove0x = (str: string): string => {
     return str
   }
 }
+
+const wait = (milliseconds) => {
+  return new Promise((resolve) => setTimeout(resolve, milliseconds))
+}
+
+export async function sendTxWithITX(
+  itx: JsonRpcProvider,
+  signer: Signer,
+  to: string,
+  data: string,
+  gas: string,
+  iteration = 0
+): Promise<string> {
+  try {
+    const tx = {
+      to,
+      data,
+      gas: BigNumber.from(gas)
+        .add(50000)
+        .toString(), // A little extra gas since we fetched it using estimateGas
+    }
+
+    const relayTransactionHashToSign = keccak256(
+      defaultAbiCoder.encode(
+        ['address', 'bytes', 'uint', 'uint'],
+        [tx.to, tx.data, tx.gas, (await itx.getNetwork()).chainId]
+      )
+    )
+    const signature = await signer.signMessage(
+      arrayify(relayTransactionHashToSign)
+    )
+
+    return await itx.send('relay_sendTransaction', [tx, signature])
+  } catch (e) {
+    // Take into account JSON RPC errors that may arise (e.g. rate limiting)
+
+    if (iteration === RETRY_LIMIT) {
+      throw e
+    }
+    await wait(2000)
+
+    // Let's try to send it again.
+    return sendTxWithITX(itx, signer, to, data, gas, iteration + 1)
+  }
+}
+export async function itxWaitForTx(
+  itx: JsonRpcProvider,
+  relayTransactionHash: string
+): Promise<TransactionReceipt> {
+  let iteration = 0
+
+  while (true) {
+    try {
+      // fetch the latest ethereum transaction hashes
+      const statusResponse = await itx.send('relay_getTransactionStatus', [
+        relayTransactionHash,
+      ])
+
+      // check each of these hashes to see if their receipt exists and
+      // has confirmations
+      for (const response of statusResponse.length) {
+        const hashes = response
```

Comment:
Could set the name in the initial for/of, although I'd suggest broadcast as a better name since the object contains more than hashes.

```
for (const broadcast of statusResponse) {
```

Reply:
changed